pallets/messages/src/lib.rs

```diff
@@ -337,27 +362,81 @@ impl<T: Config> Pallet<T> {
 			.map_err(|_| Error::<T>::InvalidMessageSourceAccount)?)
 	}
 
-	/// Gets a messages for a given schema-id and block-number.
+	/// Retrieve the messages for a particular intent and block range (paginated)
+	pub fn get_messages_by_intent_id(
+		intent_id: IntentId,
+		pagination: BlockPaginationRequest,
+	) -> BlockPaginationResponse<MessageResponseV2> {
+		let mut response = BlockPaginationResponse::new();
+
+		// Request Validation
+		if !pagination.validate() {
+			return response
+		}
+
+		// Schema Fetch and Check
+		let intent = match <T>::SchemaProvider::get_intent_by_id(intent_id) {
+			Some(intent) => intent,
+			None => return response,
+		};
+
+		let mut from_index: u32 = pagination.from_index;
+
+		let _ = (pagination.from_block..pagination.to_block).try_for_each(|block_number| {
```

Comment:
I think we can improve the readability by using a `for block_number in pagination.from_block..pagination.to_block` since we are mutating external variables

for block_number in pagination.from_block..pagination.to_block